server.go

```diff
@@ -0,0 +1,207 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/gorilla/mux"
+	emailLib "github.com/jordan-wright/email"
+	_ "github.com/lib/pq"
+)
+
+const (
+	contentType     = "Content-Type"
+	jsonContentType = "application/json;charset=UTF-8"
+)
+
+func NewServer(httpAddr string, db *sql.DB, emailPool *emailLib.Pool) *http.Server {
+	// TODO - Add logging middleware
+	// TODO - Add secure headers middleware
+	r := mux.NewRouter()
+
+	r.HandleFunc("/api/v1/email", CreateEmailAccountHandler(db)).Methods("POST")
+	r.HandleFunc("/api/v1/email/{id}/send", SendEmailHandler(db, emailPool)).Methods("POST")
+	r.HandleFunc("/api/v1/email/bulksend", SendBulkEmailHandler(db, emailPool)).Methods("POST")
+	http.Handle("/", r)
+
+	return &http.Server{
+		Addr:    httpAddr,
+		Handler: r,
+	}
+}
+
+func readReqBody(r *http.Request) ([]byte, error) {
+	defer r.Body.Close()
+	body, err := ioutil.ReadAll(io.LimitReader(r.Body, 1048576))
+	if err != nil {
+		log.Errorf("Error occured when reading r.Body: %s", err)
+		return []byte{}, err
+	}
+	return body, nil
+}
+
+type ErrorResponse struct {
+	Error string `json:"error"`
+}
+
+// Custom version of http.Error to support json error messages
+func ErrorRespond(w http.ResponseWriter, errMsg string, code int) {
+	resp := &ErrorResponse{Error: errMsg}
+
+	w.Header().Set(contentType, jsonContentType)
+	w.Header().Set("X-Content-Type-Options", "nosniff")
```

Comment:
@ajvb Interesting; what does this do?

Reply:
This really should go everywhere we do `w.Header().Set(contentType, jsonContentType)` - just tells clients/proxies that the content type should not be changed (via sniffing).